src/utils/useSessionCount.ts

```diff
@@ -0,0 +1,42 @@
+import AsyncStorage from '@react-native-community/async-storage'
+import { useEffect } from 'react'
+
+const SESSION_COUNT_KEY = '@session-count'
+
+const getSessionCount = async () => {
+  const sessionCount = await AsyncStorage.getItem(SESSION_COUNT_KEY)
+  return sessionCount ? parseInt(sessionCount, 10) : null
+}
+
+export const incrementSessionCount = async () => {
+  const sessionCount = await getSessionCount()
+  if (sessionCount) {
+    await AsyncStorage.setItem(SESSION_COUNT_KEY, (sessionCount + 1).toString())
+  } else {
+    await AsyncStorage.setItem(SESSION_COUNT_KEY, (1).toString())
+  }
+}
+
+/**
+ * Invokes `callback` every `frequency` sessions
+ * @param callback
+ * @param frequency
+ * @param startAt which session to start at
+ */
+const useSessionCount = (
```

Comment:
It seems like this guy is really sensitive to a non-memoized callback being passed in - if we're chilling on a % 0 and the callback changes once per render or something, I believe we'll call `work` multiple times? Maybe we need to track some state so that we only call work once per start value?

Reply:
good callout.